### PR TITLE
Update Vue editors demos to composition API style

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

**Comment by alexanderPolosatov** [comment]
Let's do not create PRs with equal names(name is equal with [this PR](https://github.com/DevExpress/devextreme-demos/pull/2701))

File: JSDemos/Demos/TagBox/GroupedItems/Vue/Group.vue [hunk 1]
```diff
@@ -4,15 +4,12 @@
...
-  props: {
-    itemData: {
-      type: Object,
-      default: () => {},
-    },
-  },
-};
+<script setup lang="ts">
+withDefaults(defineProps<{
+  itemData?: object
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/TagBox/GroupedItems/Vue/Group.vue]: in VSCode I see error "Parsing error: Unexpected token ? eslint".
Can we check all this errors for all changes?

Comment (GoodDayForSurf) [reply JSDemos/Demos/TagBox/GroupedItems/Vue/Group.vue]: i fixed it by update eslint

**Review by alexanderPolosatov — DISMISSED** [review]
changes in demos looks ok
But I suggest require approve of infrastructure changes(@dxvladislavvolkov )

File: JSDemos/Demos/Calendar/Overview/Vue/App.vue [hunk 2]
```diff
@@ -100,89 +100,82 @@
...
+];
+const weekNumberRules = ['auto', 'firstDay', 'firstFourDays', 'fullWeek'];
+
+function isWeekend(date) {
+  const day = date.getDay();
+  return day === 0 || day === 6;
+}
+
+function setMinDate({ value }) {
+  minDateValue.value = value
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/Calendar/Overview/Vue/App.vue:138]: minDateValue declared as null. it leads to TS error 'Type 'Date | null' is not assignable to type 'null'.'

**Review by dxvladislavvolkov — DISMISSED** [review]

File: JSDemos/Demos/DropDownBox/MultipleSelection/Vue/App.vue [hunk 3]
```diff
@@ -76,62 +76,46 @@ import {
...
-};
+  });
+}
+
+function syncTreeViewSelection() {
+  const value = treeBoxValue.value;
+
+  if (treeView) {
+    if (value === null) {
+      treeView.unselectAll();
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/DropDownBox/MultipleSelection/Vue/App.vue:107]: TS error

File: JSDemos/Demos/DropDownBox/MultipleSelection/Vue/App.vue [hunk 4]
```diff
@@ -25,8 +24,8 @@
               selection-mode="multiple"
               show-check-boxes-mode="normal"
               display-expr="name"
-              @content-ready="syncTreeViewSelection($event)"
-              @item-selection-changed="treeView_itemSelectionChanged($event)"
+              @content-ready="treeViewContentReady($event)"
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/DropDownBox/MultipleSelection/Vue/App.vue:27]: TS error

File: JSDemos/Demos/TagBox/Overview/Vue/App.vue [hunk 5]
```diff
@@ -117,41 +117,24 @@
...
-  },
-};
+const editableProducts = ref(simpleProducts.slice());
+
+const value = ref([1, 2]);
+const onCustomItemCreating = ref((args) => {
+  const newValue = args.text;
+  const isItemInDataSource = editableProducts.value.some((item) => item === newValue);
+  if (!isItemInDataSource) {
+    editableProducts.value = [newValue, ...editableProducts];
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/TagBox/Overview/Vue/App.vue]: TS error

**Review by alexanderPolosatov — COMMENTED** [review]
We have several TS errors after start using of TS. Should we fix this?

**Review by alexanderPolosatov — APPROVED** [review]

File: JSDemos/Demos/TagBox/Overview/Vue/App.vue (hunk 5 above)
Comment (GoodDayForSurf) [reply JSDemos/Demos/TagBox/Overview/Vue/App.vue]: we will fix ts issues in separate task

**Pull request merged**